### ISSUE TITLE
Dependencies not found for projects that contains uppercase letters

### ISSUE THREAD

**Issue by artrayd** [vuejs/vue-cli#4251]
### Version
3.9.2

### Environment info
```
  System:
    OS: macOS 10.14.1
  Binaries:
    Node: 10.8.0 - /usr/local/bin/node
    Yarn: Not Found
    npm: 6.9.2 - /usr/local/bin/npm
  Browsers:
    Chrome: 75.0.3770.100
  npmGlobalPackages:
    @vue/cli: 3.9.2
```

### Steps to reproduce
Vue UI

Then create "ProjectName", with big "P" and "N"

Tasks --> Serve

### What is expected?
The project will be served

### What is actually happening?
These dependencies were not found:

/Users/newuser/Documents/projects/wheel/ProjectName/node_modules/webpack-dev-server/client/index.js?http://10.0.0.4:8080/sockjs-node in multi ../ProjectName/node_modules/webpack-dev-server/client?http://10.0.0.4:8080/sockjs-node ../ProjectName/node_modules/webpack/hot/dev-server.js ./src/main.js, multi ../ProjectName/node_modules/webpack-dev-server/client?http://localhost ../ProjectName/node_modules/webpack/hot/dev-server.js ../ProjectName/node_modules/webpack-dev-server/client?http://10.0.0.4:8080/sockjs-node ./src/main.js
/Users/newuser/Documents/projects/wheel/ProjectName/node_modules/webpack-dev-server/client/index.js?http://localhost in multi ../ProjectName/node_modules/webpack-dev-server/client?http://localhost ../ProjectName/node_modules/webpack/hot/dev-server.js ../ProjectName/node_modules/webpack-dev-server/client?http://10.0.0.4:8080/sockjs-node ./src/main.js
/Users/newuser/Documents/projects/wheel/ProjectName/node_modules/webpack/hot/dev-server.js in multi ../ProjectName/node_modules/webpack-dev-server/client?http://10.0.0.4:8080/sockjs-node ../ProjectName/node_modules/webpack/hot/dev-server.js ./src/main.js, multi ../ProjectName/node_modules/webpack-dev-server/client?http://localhost ../ProjectName/node_modules/webpack/hot/dev-server.js ../ProjectName/node_modules/webpack-dev-server/client?http://10.0.0.4:8080/sockjs-node ./src/main.js
To install them, you can run: npm install --save /Users/newuser/Documents/projects/wheel/ProjectName/node_modules/webpack-dev-server/client/index.js?http://10.0.0.4:8080/sockjs-node /Users/newuser/Documents/projects/wheel/ProjectName/node_modules/webpack-dev-server/client/index.js?http://localhost /Users/newuser/Documents/projects/wheel/ProjectName/node_modules/webpack/hot/dev-server.js

This relative module was not found:

./src/main.js in multi ../ProjectName/node_modules/webpack-dev-server/client?http://10.0.0.4:8080/sockjs-node ../ProjectName/node_modules/webpack/hot/dev-server.js ./src/main.js, multi ../ProjectName/node_modules/webpack-dev-server/client?http://localhost ../ProjectName/node_modules/webpack/hot/dev-server.js ../ProjectName/node_modules/webpack-dev-server/client?http://10.0.0.4:8080/sockjs-node ./src/main.js

---
Caused by this issue:
https://github.com/vuejs/vue-cli/issues/2547

**Comment by tripflex** [comment]
I can confirm this is an issue as well

**Comment by sodatea** [comment]
It's not caused by that issue but it's a reason that we don't want to implement uppercase project name support. It brings so many edge cases that it becomes a maintenance burden for us.

**Comment by artrayd** [comment]
> It's not caused by that issue but it's a reason that we don't want to implement uppercase project name support. It brings so many edge cases that it becomes a maintenance burden for us.

I understand that, so please add error or warning state with small notice about that capital letters aren't supported or just convert it to lowercase automatically. Because today users can easily create project that will not work because of it and it's pretty hard to figure out the reason why it's not working :)

**Comment by sodatea** [comment]
Yeah, I'm thinking about whether to disallow the creation of projects with uppercase names in the UI or convert it to lowercase.

In this case, the bug is caused by the UI logic missed one case conversion. But I don't know how many more edge cases there would be if we fully support such use cases.

**Comment by motla** [comment]
issue is still here 😉
